rsc/src/main/scala/rsc/scan/Scanner.scala

```diff
@@ -447,8 +447,9 @@ final class Scanner private (
   }
 
   private def emitIdOrKeyword(): Unit = {
-    val token = keywords.getOrElse(lexeme, ID)
-    val value = if (token == ID) lexeme else null
+    val element = lexeme
```

Comment:
each call to lexeme allocates a new String, separating this out halves the number of strings generates.